packages/component-library-angular/src/button-link/ButtonLink.spec.ts

```diff
@@ -0,0 +1,190 @@
+      const { container } = await render(
+        '<a href="https://example.com" [appearance]="appearance" utrecht-button-link></a>',
+        {
+          declarations: [UtrechtButtonLinkAttr],
+          componentProperties: {
+            appearance: 'secondary-action-button',
+          },
+        },
+      );
+      const buttonLink = container.querySelector('.utrecht-button-link');
+
+      expect(buttonLink).toHaveClass('utrecht-button-link--secondary-action');
+    });
+
+    it('renders a design system BEM `utrecht-button-link--subtle-action` class', async () => {
+      const { container } = await render(
+        '<a href="https://example.com" [appearance]="appearance" utrecht-button-link></a>',
+        {
+          declarations: [UtrechtButtonLinkAttr],
+          componentProperties: {
+            appearance: 'subtle-button',
+          },
+        },
+      );
+      const buttonLink = container.querySelector('.utrecht-button-link');
+
+      expect(buttonLink).toHaveClass('utrecht-button-link--subtle-action');
+    });
+    it('can not renders a design system BEM `utrecht-button-link--primary-action` class by default', async () => {
+      const { container } = await render('<a href="https://example.com"  utrecht-button-link></a>', {
+        declarations: [UtrechtButtonLinkAttr],
+      });
+      const buttonLink = container.querySelector('.utrecht-button-link');
+
+      expect(buttonLink).not.toHaveClass('utrecht-button-link--primary-action');
+    });
+    it('can not renders a design system BEM `utrecht-button-link--secondary-action` class be default', async () => {
+      const { container } = await render('<a href="https://example.com" utrecht-button-link></a>', {
+        declarations: [UtrechtButtonLinkAttr],
+      });
+      const buttonLink = container.querySelector('.utrecht-button-link');
+
+      expect(buttonLink).not.toHaveClass('utrecht-button-link--secondary-action');
+    });
+
+    it('can not renders a design system BEM `utrecht-button-link--subtle-action` class by default', async () => {
+      const { container } = await render('<a href="https://example.com" utrecht-button-link></a>', {
+        declarations: [UtrechtButtonLinkAttr],
+      });
+      const buttonLink = container.querySelector('.utrecht-button-link');
+
+      expect(buttonLink).not.toHaveClass('utrecht-button-link--subtle-action');
+    });
+  });
+
+  describe('Button Link `external`', () => {
+    it('prevents sharing referer information', async () => {
+      await render('<a utrecht-button-link [external]="external" href="https://example.com">Link</a>', {
+        declarations: [UtrechtButtonLinkAttr],
+        componentProperties: {
+          external: true,
+        },
+      });
+
+      const buttonLink = screen.getByRole('link', { name: 'Link' });
+
+      expect(buttonLink).toHaveAttribute('rel', 'external noopener noreferer');
```

Comment:
The order of the relationship tokens are not relevant, so I would try to test for this in another way. For example using the attribute value selector that works specifically for whitespace separated lists: `a[rel~="external" i]`.

For more specific error messages, I would make a separate `expect()` for every value.

The `prevents sharing referer info` should only check for `noreferer` by the way.